include/drivers/gpio.h

```diff
@@ -566,7 +652,21 @@ static inline int gpio_pin_configure(struct device *port, u32_t pin,
 		 || (flags & GPIO_OUTPUT) != 0,
 		 "Output needs to be enabled to be initialized low or high");
 
-	return gpio_config(port, GPIO_ACCESS_BY_PIN, pin, flags);
+	ret = gpio_config(port, GPIO_ACCESS_BY_PIN, pin, flags);
+
+	if (ret == 0) {
```

Comment:
This block seems to change the API contract.  In the merged design (and still documented) `gpio_pin_configure()` does not interpret interrupt configuration.  Thus pin configuration and interrupt configuration are completely separated.

With this implementation configuring a pin appears to also configure the interrupts.  That seems to be mixing capabilities that were intended (as I understood it) to be kept separate in the new API.

Reply:
This is required to keep backward compatibility with the existing application code. Will be removed in the future. As I have mentioned already, I'm not really sure how to handle deprecation of a feature like this one so suggestions are welcome.